src/lib/statsig/statsig.tsx

```diff
@@ -98,16 +94,23 @@ export function logEvent<E extends keyof LogEvents>(
   }
 }
 
-export function useGate(gateName: Gate): boolean {
-  const {isLoading, value} = useStatsigGate(gateName)
-  if (isLoading) {
-    // This should not happen because of waitForInitialization={true}.
-    console.error('Did not expected isLoading to ever be true.')
+export function useGate(): (gateName: Gate) => boolean {
+  const cache = React.useRef<Map<Gate, boolean>>()
+  if (cache.current === undefined) {
+    cache.current = new Map()
   }
```

Comment:
Maybe silly question, but why not set the initial value in the hook call?

Reply:
Oh lawl just saw this comment https://github.com/bluesky-social/social-app/pull/3498/files#r1568218021

Reply:
Can you show the code you mean?